catched panic of add index runDDLJob

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
[2024/07/09 23:26:58.243 +08:00] [INFO] [backend.go:354] ["engine close completed"] [category=ddl-ingest] [engineTag=sbtest1:94] [engineUUID=24fe2a81-4dc7-5380-a88a-0a7cb3f2523d] [takeTime=9.871228672s] []
[2024/07/09 23:26:58.247 +08:00] [ERROR] [misc.go:114] ["panic in the recoverable goroutine"] [label=ddl-worker] [funcInfo="worker 3, tp add index runDDLJob"] [r="send on closed channel"] [stack="github.com/pingcap/tidb/pkg/util.Recover\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/util/misc.go:118\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:914\nruntime.selectgo\n\t/usr/local/go/src/runtime/select.go:515\ngithub.com/pingcap/tidb/pkg/lightning/backend/local.(*Engine).addSST\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/lightning/backend/local/engine.go:816\ngithub.com/pingcap/tidb/pkg/lightning/backend/local.(*Writer).addSST\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/lightning/backend/local/engine.go:1355\ngithub.com/pingcap/tidb/pkg/lightning/backend/local.(*Writer).flushKVs\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/lightning/backend/local/engine.go:1343\ngithub.com/pingcap/tidb/pkg/lightning/backend/local.(*Writer).flush\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/lightning/backend/local/engine.go:1255\ngithub.com/pingcap/tidb/pkg/lightning/backend/local.(*Writer).Close\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/lightning/backend/local/engine.go:1301\ngithub.com/pingcap/tidb/pkg/ddl/ingest.(*engineInfo).closeWriters\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/ingest/engine.go:198\ngithub.com/pingcap/tidb/pkg/ddl/ingest.(*engineInfo).Clean\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/ingest/engine.go:121\ngithub.com/pingcap/tidb/pkg/ddl/ingest.(*litBackendCtx).FinishAndUnregisterEngines\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/ingest/engine_mgr.go:114\ngithub.com/pingcap/tidb/pkg/ddl/ingest.(*litBackendCtxMgr).Unregister\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/ingest/backend_mgr.go:234\ngithub.com/pingcap/tidb/pkg/ddl.convertAddIdxJob2RollbackJob\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/rollingback.go:95\ngithub.com/pingcap/tidb/pkg/ddl.runReorgJobAndHandleErr\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/index.go:1069\ngithub.com/pingcap/tidb/pkg/ddl.runIngestReorgJobDist\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/index.go:943\ngithub.com/pingcap/tidb/pkg/ddl.doReorgWorkForCreateIndex\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/index.go:893\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).onCreateIndex\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/index.go:759\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).runDDLJob\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/ddl_worker.go:1265\ngithub.com/pingcap/tidb/pkg/ddl.(*worker).HandleDDLJobTable\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/ddl_worker.go:950\ngithub.com/pingcap/tidb/pkg/ddl.(*jobScheduler).runOneJobStep\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/job_table.go:600\ngithub.com/pingcap/tidb/pkg/ddl.(*jobScheduler).deliveryJob.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/ddl/job_table.go:532\ngithub.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).Run.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/util/wait_group_wrapper.go:157"]
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/ddl/ingest/engine.go
+++ b/pkg/ddl/ingest/engine.go
@@ -110,6 +110,12 @@ func (ei *engineInfo) Clean() {
 	if ei.openedEngine == nil {
 		return
 	}
+	err := ei.closeWriters()
+	if err != nil {
+		logutil.Logger(ei.ctx).Error(LitErrCloseWriterErr, zap.Error(err),
+			zap.Int64("job ID", ei.jobID), zap.Int64("index ID", ei.indexID))
+	}
+
 	indexEngine := ei.openedEngine
 	closedEngine, err := indexEngine.Close(ei.ctx)
 	if err != nil {
@@ -118,11 +124,6 @@ func (ei *engineInfo) Clean() {
 		return
 	}
 	ei.openedEngine = nil
-	err = ei.closeWriters()
-	if err != nil {
-		logutil.Logger(ei.ctx).Error(LitErrCloseWriterErr, zap.Error(err),
-			zap.Int64("job ID", ei.jobID), zap.Int64("index ID", ei.indexID))
-	}
 	// Here the local intermediate files will be removed.
 	err = closedEngine.Cleanup(ei.ctx)
 	if err != nil {

